### PR TITLE
processing data in models

### PR DESCRIPTION
**Pull request by Surafels**
Hello 👋

Implemented tasks:

- [ ] Create models

- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png).

- Set up associations between models.
- [ ] Use models to insert data

- [ ] Create custom methods

1. Users

- A method that returns the 3 most recent posts for a given user.

2. Posts

- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.

3. Comments

- A method that updates the comments counter for a post.

4. Likes

- A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_user_posts_counter
+    user.update(post_counter: user.post.count)
+  end
```
Comment (DammyShittu) [inline app/models/post.rb]: - [ ] This should be `author.update(...)`, not `user.update(...)` as a post belongs to an author, not a user.

- **[OPTIONAL]** I'd suggest you add an `after_create` active record callback for this method.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  after_create :update_user_comment_counter
+  def update_user_comment_counter
+    user.update(comment_counter: post.comments.count)
```
Comment (DammyShittu) [inline app/models/comment.rb]: - [ ] The `post` is what has a comments counter, not `user`.

- [ ] Kindly run a migration to add `comments_counter` to the posts table. This method will raise errors if you call it as there is no such column on your table.

  image

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  after_create :update_likes_counter
+  def update_likes_counter
+    post.update(likes_count: post.likes.count)
```
Comment (DammyShittu) [inline app/models/like.rb]: - [ ] This should be `likes_counter`, not `likes_count` based on your schema.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Surafels,

## Highlights
✔️ Working program

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @Surafels 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on fixing all the issues mentioned by the previous reviewer. 💯 

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**